### PR TITLE
DBZ-2167 Use legacy parser for testing purposes only

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2167

### CODE REVIEW

File: debezium-core/src/test/java/io/debezium/util/parser/TokenStream.java [hunk 1]
```diff
@@ -0,0 +1,2592 @@
...
+import io.debezium.annotation.Immutable;
+import io.debezium.annotation.NotThreadSafe;
+import io.debezium.function.BooleanConsumer;
+import io.debezium.text.ParsingException;
+import io.debezium.text.Position;
+import io.debezium.text.XmlCharacters;
+import io.debezium.util.Strings;
+
+/**
+ * A copy of {@link io.debezium.text.TokenStream} class that provides additional methods necessary for test parser.
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/util/parser/TokenStream.java:25]: Ugh :) Why again do we need `TokenStream` in main? And if it's truly needed over there (and not test-only), can't we extend?

Comment (jpechane) [reply debezium-core/src/test/java/io/debezium/util/parser/TokenStream.java:25]: It is used in `TableIdParser` and in `Strings`.

I was attemptin to do the extension first but that would need not insignificant changes including even generfication of the class which I thought is unnecessary complication. Hence I followed the path of the least resistence and made the copy so the original tokenizer is not polluted with SQL specific stuff.

File: debezium-core/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java [hunk 2]
```diff
@@ -29,11 +29,11 @@
 import io.debezium.doc.FixFor;
 import io.debezium.kafka.KafkaCluster;
 import io.debezium.relational.Tables;
-import io.debezium.relational.ddl.DdlParserSql2003;
-import io.debezium.relational.ddl.LegacyDdlParser;
 import io.debezium.text.ParsingException;
 import io.debezium.util.Collect;
 import io.debezium.util.Testing;
+import io.debezium.util.parser.DdlParserSql2003;
+import io.debezium.util.parser.LegacyDdlParser;
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java:36]: How about moving the affected tests to the MySQL connector module then, so we could get rid of the Legacy parser impl altogether?

Comment (jpechane) [reply debezium-core/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java]: @gunnarmorling Done. I am not sure if it is structurally clean but as a pragmatic solution it works.

File: debezium-connector-mysql/src/test/java/io/debezium/relational/history/AbstractDatabaseHistoryTest.java [hunk 3]
```diff
@@ -94,8 +94,8 @@ protected Tables recover(long pos, int entry) {
     @Test
     public void shouldRecordChangesAndRecoverToVariousPoints() {
         record(01, 0, "CREATE TABLE foo ( first VARCHAR(22) NOT NULL );", all, t3, t2, t1, t0);
-        record(23, 1, "CREATE TABLE\\nperson ( name VARCHAR(22) NOT NULL );", all, t3, t2, t1);
-        record(30, 2, "CREATE TABLE address\\n( street VARCHAR(22) NOT NULL );", all, t3, t2);
+        record(23, 1, "CREATE TABLE\nnperson ( name VARCHAR(22) NOT NULL );", all, t3, t2, t1);
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/relational/history/AbstractDatabaseHistoryTest.java:97]:
```suggestion
        record(23, 1, "CREATE TABLE\nperson ( name VARCHAR(22) NOT NULL );", all, t3, t2, t1);
```

One "n" too much?

File: debezium-core/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java (hunk 2 above)
Comment (gunnarmorling) [reply debezium-core/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java]: Yes, it's a bit of a blurry line. But maintaining a whole parser impl only for testing seems like too much of an overhead to me. I think it's the better approach now.

File: debezium-connector-mysql/src/test/java/io/debezium/relational/history/AbstractDatabaseHistoryTest.java (hunk 3 above)
Comment (jpechane) [reply debezium-connector-mysql/src/test/java/io/debezium/relational/history/AbstractDatabaseHistoryTest.java]: `n`s are for free so I probably wanted to be generous with them before Christmas ;-)

Comment (gunnarmorling) [reply debezium-connector-mysql/src/test/java/io/debezium/relational/history/AbstractDatabaseHistoryTest.java]: LOL :)

**Comment by gunnarmorling** [comment]
LGTM, awaiting CI. Nice add/removal line count ratio :)

**Comment by jpechane** [comment]
@gunnarmorling Done, test failure seems to be even rarer instance of https://issues.redhat.com/browse/DBZ-2231

**Comment by gunnarmorling** [comment]
Applied, thx! Let's keep an eye on that failure and try and fix it if it's persistent.

**Pull request merged**